src/ln/router.rs

```diff
@@ -96,6 +97,54 @@ impl std::fmt::Display for DirectionalChannelInfo {
 	}
 }
 
+impl Writeable for DirectionalChannelInfo {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		self.src_node_id.write(writer)?;
+		self.last_update.write(writer)?;
+		self.enabled.write(writer)?;
+		self.cltv_expiry_delta.write(writer)?;
+		self.htlc_minimum_msat.write(writer)?;
+		self.fee_base_msat.write(writer)?;
+		self.fee_proportional_millionths.write(writer)?;
+		match &self.last_update_message {
+			&None => 0u8.write(writer)?,
+			&Some(ref msg) => {
+				1u8.write(writer)?;
+				msg.write(writer)?;
+			}
+		}
+		Ok(())
+	}
+}
+
+impl<R: ::std::io::Read> Readable<R> for DirectionalChannelInfo {
+	fn read(reader: &mut R) -> Result<DirectionalChannelInfo, DecodeError> {
+		let src_node_id = Readable::read(reader)?;
+		let last_update = Readable::read(reader)?;
+		let enabled = Readable::read(reader)?;
+		let cltv_expiry_delta = Readable::read(reader)?;
+		let htlc_minimum_msat = Readable::read(reader)?;
+		let fee_base_msat = Readable::read(reader)?;
+		let fee_proportional_millionths = Readable::read(reader)?;
+		let last_update_message = match <u8 as Readable<R>>::read(reader)? {
+			0 => None,
+			1 => Some(msgs::ChannelUpdate::read(reader)?),
```

Comment:
Maybe we should just implement Readable/Writeable for Option<T : Readable/Writeable>? There's a bunch of places that would simplify things.

Reply:
Hey right working on it